### PR TITLE
Use Github Actions to push ECR on new tag, upgrade Chamber

### PR DESCRIPTION
**Pull request by mbarrien**
Adds a Github Action to build and push Docker images to ECR.  This replaces the current use of Docker Hub. Despite this being a public repo, the ECR repo is only accessible from within our organization. This will also remove our last use of Docker Hub. If we do intend to keep this image public, I'd move us to Github Packages instead.

While here, upgrades the version of Chamber to the latest.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        uses: docker/metadata-action@v3
+        with:
+          # list of Docker images to use as base name for tags
+          images: ${{ secrets.ECR_REPO }}
+          # generate Docker tags based on the following events/attributes
+          tags: |
+            type=ref,event=branch,prefix=branch-
+            type=ref,event=tag
+            type=sha
+      - name: Set up Docker Buildx
```
Comment (atolopko-czi) [inline .github/workflows/build.yml:32]:
```suggestion
      - name: Set up Docker Build
```

Comment (atolopko-czi) [reply .github/workflows/build.yml:32]: Or is `Buildx` correct?, I see it in next line, so maybe not a typo.

**Review by atolopko-czi — COMMENTED** [review]
Thanks for doing this. We'll also need to create a new Task Def to replace [this one](https://us-west-2.console.aws.amazon.com/ecs/home?region=us-west-2#/taskDefinitions/meta-prod-datadog-backup/2), and test the associated ECS [schedule service](https://us-west-2.console.aws.amazon.com/ecs/home?region=us-west-2#/clusters/meta-prod-dpt-ecs/scheduledTasks/meta-prod-datadog-backup).

While this backup service seems like a good idea, I'm not sure we've every _needed it_.  However, if deemed valuable, this seems like something that could be managed by Central Infra.  It's basically version control for Datadog (literally so, since it's backs up Datadog state to a git repo).

File: .github/workflows/build.yml (hunk 1 above)
Comment (mbarrien) [reply .github/workflows/build.yml:32]: Buildx is correct. https://docs.docker.com/buildx/working-with-buildx/

**Comment by mbarrien** [comment]
Our solution so far has been to encourage teams to manage Datadog settings via Terraform and the Datadog provider, which inverts it so that we have infra-as-code instead of backing up the output of infra-as-clicks-in-the-UI. I do know however that the TF Datadog experience can be clunky.

I have PRs ready to replace the Task Def once we decide where the Docker image should reside. The schedule service should not need changing since it just refers to the task def.

Note that this repo is public readable; make sure that links you're including in the Github comments are appropriate for public consumption.

File: .github/workflows/build.yml (hunk 1 above)
Comment (atolopko-czi) [reply .github/workflows/build.yml:32]: Ah! Thx.

**Review by atolopko-czi — APPROVED** [review]
I don't see any need to have the built Docker image be public, so let's go with ECR.

**Comment by atolopko-czi** [comment]
> Our solution so far has been to encourage teams to manage Datadog settings via Terraform and the Datadog provider, which inverts it so that we have infra-as-code instead of backing up the output of infra-as-clicks-in-the-UI. I do know however that the TF Datadog experience can be clunky.

Good to know--will assess TF-based Datadog definitions next time we need to make significant monitoring changes.

**Pull request merged**